src/Core/CodeModel/CSharpExpressionWriter.cs

```diff
@@ -16,84 +16,76 @@
-        private const int PrecUnary = 13;
-        private const int PrecMultplicative = 12;
-        private const int PrecAdditive = 11;
-        private const int PrecShift = 10;
-        private const int PrecRelational = 9;
-        private const int PrecEquality = 8;
-        private const int PrecBinAnd = 7;
-        private const int PrecBitXor = 6;
-        private const int PrecBitOr = 5;
-        private const int PrecLogicalAnd = 4;
-        private const int PrecLogicalOr = 3;
-        private const int PrecConditional = 2;
-        private const int PrecAssignment = 1;
-        private const int PrecBase = 0;
-
-        private static Dictionary<CodeOperatorType, int> operatorPrecedence = new Dictionary<CodeOperatorType, int>
-        {
-            { CodeOperatorType.Complement, PrecUnary },
-            { CodeOperatorType.Not, 13 },
-
-            { CodeOperatorType.Mul, 12 },
-            { CodeOperatorType.Div, 12 },
-            { CodeOperatorType.Mod, 12 },
-
-            { CodeOperatorType.Add, 11 },
-            { CodeOperatorType.Sub, 11 },
-
-            { CodeOperatorType.Shl, 10 },
-            { CodeOperatorType.Shr, 10 },
-
-            { CodeOperatorType.Lt, 9 },
-            { CodeOperatorType.Gt, 9 },
-            { CodeOperatorType.Le, 9 },
-            { CodeOperatorType.Ge, 9 },
-
-            { CodeOperatorType.Equal, 8 },
-            { CodeOperatorType.NotEqual, 8 },
-            { CodeOperatorType.IdentityEquality, 8 },
-            { CodeOperatorType.IdentityInequality, 8 },
-            { CodeOperatorType.Is, 8 },
-
-            { CodeOperatorType.BitAnd, 7 },
-
-            { CodeOperatorType.BitXor, 6 },
-
-            { CodeOperatorType.BitOr, 5 },
-
-            { CodeOperatorType.LogAnd, 4 },
-
-            { CodeOperatorType.LogOr, 3 },
-             
-            { CodeOperatorType.Conditional, 2 },
-
-            { CodeOperatorType.Assign, 1 },
-            { CodeOperatorType.AddEq, 1 },
-            { CodeOperatorType.SubEq, 1 },
-            { CodeOperatorType.ModEq, 1 },
-            { CodeOperatorType.MulEq, 1 },
-            { CodeOperatorType.AndEq, 1 },
-            { CodeOperatorType.OrEq, 1 },
-            { CodeOperatorType.DivEq, 1 },
-            { CodeOperatorType.ShlEq, PrecAssignment },
-            { CodeOperatorType.ShrEq, PrecAssignment },
-            { CodeOperatorType.XorEq, PrecAssignment },
+
+        enum Prec
+        {
+            Base = 0,
+            Assignment = 1,
+            Conditional = 2,
+            LogicalOr = 3,
+            LogicalAnd = 4,
+            BitOr = 5,
+            BitXor = 6,
+            BinAnd = 7,
+            Equality = 8,
+            Relational = 9,
+            Shift = 10,
+            Additive = 11,
+            Multplicative = 12,
+            Unary = 13,
+            Postfix = 14,
+            Primary = 15,
+
+        }
+
+        private static Prec operatorPrecedence(CodeOperatorType op) => op switch
```

Comment:
I'm curious why you changed this from a Dictionary lookup?

Reply:
t is slightly faster and easier to maintain should the precedences ever need to be amended. 
It looks more concise and does the job. Aesthetics, sorry.